storage/memory/memory_test.go

```diff
@@ -352,6 +352,20 @@ func TestObjectsFilter(t *testing.T) {
 			p:    testutil.MustBuildPredicate(t, `"_predicate"@[]`),
 			want: map[string]int{`"meet"@[2021-04-10T04:21:00Z]`: 1},
 		},
+		{
+			id:   "FILTER isImmutable predicate",
+			lo:   &storage.LookupOptions{FilterOptions: &filter.StorageOptions{Operation: filter.IsImmutable, Field: filter.PredicateField}},
+			s:    testutil.MustBuildNodeFromStrings(t, "/u", "john"),
+			p:    testutil.MustBuildPredicate(t, `"parent_of"@[]`),
+			want: map[string]int{"/u<paul>": 1},
+		},
+		{
+			id:   "FILTER isImmutable object",
+			lo:   &storage.LookupOptions{FilterOptions: &filter.StorageOptions{Operation: filter.IsImmutable, Field: filter.ObjectField}},
+			s:    testutil.MustBuildNodeFromStrings(t, "/_", "bn"),
+			p:    testutil.MustBuildPredicate(t, `"_predicate"@[]`),
+			want: map[string]int{`"height_cm"@[]`: 1},
+		},
```

Comment:
Could you change the name of the test struct a bit?
You could explain what is the want, e.g., 
want -> ExpectedCountByObjectName.
s -> query_subject
p-> query_predicate

Also the logic in the test body seems a bit too complex for tests.
Instead of decrementing the map counter and deleting if the count gets to 0,
WDYT of building the "got" map and calling equal on both?

instead of:
```go
for s := range ss {
  sStr := s.String()
  if _, ok := entry.want[sStr]; !ok {

  t.Fatalf("g.Subjects(%s, %s, %s) retrieved unexpected %s", p, o, entry.lo, sStr)						 
  }
  entry.want[sStr] = entry.want[sStr] - 1
  if entry.want[sStr] == 0 {
    delete(entry.want, sStr)
  }
}
```
something like:
```go
got := map[string]int{}
for o := range os {
  oStr := s.String()
  got[oStr] += 1
}
if !cmp.Equals(entry.want, got){
  t.Fatalf("g.Subjects(%s, %s, %s) retrieved unexpected %s", p, o, entry.lo, sStr)
}
```

Reply:
As we commented through chat, I can be doing this in a separate PR after this PR and the PR 154 are merged. Also, I will be letting this task as P2 for now to focus on current tasks, as we commented through chat as well.